src/Octoshift/Services/GithubApi.cs

```diff
@@ -1047,6 +1047,33 @@ mutation abortRepositoryMigration(
         }
     }
 
+    public virtual async Task<string> UploadArchiveToGithubStorage(string org, bool isMultipart, string archiveName, Stream archiveContent)
+    {
+        using HttpContent httpContent = new StreamContent(archiveContent);
+        string response;
+
+        if (isMultipart)
+        {
+            var url = $"{_apiUrl}/organizations/{org.EscapeDataString()}/gei/archive/blobs/uploads";
+
+            using var content = new MultipartFormDataContent
+            {
+                { httpContent, "archive", archiveName }
+            };
+
+            response = await _client.PostAsync(url, content);
```

Comment:
A couple of observations:
1. Even though this may work the `PostAsync` method will convert the data to JSON (basically base64 string) and will eventually create `StringContent` so I am not sure if this is the indented use here because we're now dealing with a `Stream` rather than string content. So We may want to refactor the [SendAsync](https://github.com/github/gh-gei/blob/62960fcd0f1fc00ef20e380ed52673fc8b41d47b/src/Octoshift/Services/GithubClient.cs#L140) method to also support a `MultiPartFormDataContent` and `StreamContent`. It can simply check to see if the passed in `body` is either of those and not convert it to a string content.
2. When dealing with a stream we shouldn't log the entire body as we do [here](https://github.com/github/gh-gei/blob/62960fcd0f1fc00ef20e380ed52673fc8b41d47b/src/Octoshift/Services/GithubClient.cs#L154) instead as I suggested in option 1, we can check the body and if it's a multipart form data or a stream content we can just say `BLOB` or `Binary data` instead of dumping the entire JSON encoded body!